### ISSUE TITLE
sel4testdriver causes vmfault on completion; expected or not?

### ISSUE THREAD

**Issue by bennoleslie** [seL4/sel4test#27]
When the seL4 test root server exits it causes a NULL pointer dereference.

Is this expected behavior? Discussion with some developers indicated it was not, but reading the code seems to suggest this would be expected.

The output is:

```
All is well in the universe


Caught cap fault in send phase at address 0
while trying to handle:
vm fault on code at address 0 with status 0x82000006
in thread 0xffffff80bffe9400 "sel4test-driver" at address 0
With stack:
0x6d3ea0: 0x6d3ec0
0x6d3ea8: 0x40e88c
0x6d3eb0: 0x6d3f20
0x6d3eb8: 0x7e8000
0x6d3ec0: 0x6d3f00
0x6d3ec8: 0x40e9cc
0x6d3ed0: 0x0
0x6d3ed8: 0x6d3f40
0x6d3ee0: 0x6d3f30
0x6d3ee8: 0x6d3f20
0x6d3ef0: 0x2
0x6d3ef8: 0x4045a4
0x6d3f00: 0x0
0x6d3f08: 0x40e78c
0x6d3f10: 0x0
0x6d3f18: 0x7e8000
```

There is something disconcerting about seeing `All is well in the universe` following by a VM fault and a stack dump!

Reading through the run time the comment on the exit path is:

```
    /* If the exit is never set this will try and call a NULL function
     * pointer which should result in a fault. This is as good a way as
     * any to exit the process if we don't know anything better about
     * the environment. */
    env.exit_cb(code);
```

Which, seems reasonable, but the customer experience of seeing a stack trace at the end of testing is still not ideal.

Thoughts on addressing this are (in roughly difficulty order)

* Put an additional print in sel4test to inform the user `you should expect to see a NULL pointer dereference as the root task exits now`
* Put an infinite loop at the end of testing 
* Put a board shutdown or reset at the end of testing

**Comment by ratmice** [comment]
I believe that this was fixed in 7b1f1fcb0bf9625766c52c149ded5abe9c830ca5, and I know longer see it.

Perhaps we can close this one.

**Comment by lsf37** [comment]
Correct, this is no longer happening in the logs of sel4test runs.
